funding/manager.go

```diff
@@ -2171,6 +2171,15 @@ func (f *Manager) continueFundingAccept(resCtx *reservationWithCtx,
 		f.failFundingFlow(resCtx.peer, pendingChanID, err)
 		return
 	}
+
+	// With FundingCreated sent, we can now notify Brontide to keep track
+	// of this pending open channel.
+	err = resCtx.peer.AddPendingChannel(channelID, f.quit)
```

Comment:
To fully avoid race conditions for zeroconf, I think we actually need to call `AddPendingChannel` *before* we send `funding_created`.

Otherwise the following sequence of events is possible:

1. We send `funding_created` and then immediately this thread is descheduled (before calling `AddPendingChannel`)
2. Peer sends `funding_signed`
3. Peer immediately sends `channel_ready` and `update_fee` since this is a zeroconf channel
4. Our channel manager processes the `update_fee` and force closes
5. Our `funding_created` thread gets scheduled again and we call `AddPendingChannel`

Reply:
great catch

Reply:
Nice catch!

Reply:
This leads me to think, we'd need a new method, `CancelPendingChannel` on `brontide` to cancel the channel if the funding flow fails?